src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,113 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.stream.IntStream;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity <= 0) {
+            throw new IllegalArgumentException();
+        } else {
+            elements = new Object[initialCapacity];
+        }
+    }
+
+    public ArrayList() {
+        this(DEFAULT_CAPACITY);
+    }
+
+    public void resizeIdIndex() {
+        if (elements.length == size) {
+            Object[] newElements = new Object[elements.length * 3 / 2 + 1];
+            System.arraycopy(elements, 0, newElements, 0, size);
+            elements = newElements;
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        resizeIdIndex();
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "index is negative or index is bigger than size");
+        }
+        Objects.checkIndex(index, size + 1);
+        resizeIdIndex();
+        System.arraycopy(elements, index, elements, index + 1, size - index);
+        elements[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        IntStream.range(0, list.size())
+                .mapToObj(list::get)
+                .forEach(this::add);
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public T get(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Can't get index,because index is negative or index is bigger than size");
+        }
+        Objects.checkIndex(index, size);
+        return (T) elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Can't set index, because index is negative or index is bigger than size");
+        }
```

Comment:
move **all** of the index/position checks like this to a separate private method, and add index itself to the exception message